src/framework/components/anim/binder.js

```diff
@@ -0,0 +1,263 @@
+Object.assign(pc, function () {
+
+    var AnimComponentBinder = function (animComponent, graph) {
+        this.animComponent = animComponent;
+        this.propertyLocator = new pc.AnimPropertyLocator();
+
+        if (graph) {
+            var nodes = { };
+
+            // cache node names so we can quickly resolve animation paths
+            var flatten = function (node) {
+                nodes[node.name] = {
+                    node: node,
+                    count: 0
+                };
+                for (var i = 0; i < node.children.length; ++i) {
+                    flatten(node.children[i]);
+                }
+            };
+            flatten(graph);
+
+            this.nodes = nodes;                 // map of node name -> { node, count }
+            this.activeNodes = [];              // list of active nodes
+        }
+    };
+
+    Object.assign(AnimComponentBinder.prototype, {
+        resolve: function(path) {
+            var pathSections = this.propertyLocator.decode(path);
+
+            var entityHeirarchy = pathSections[0];
+            var component = pathSections[1];
+            var propertyHeirarchy = pathSections[2];
```

Comment:
`propertyHeirarchy` -> `propertyHieirarchy `

Reply:
in general `heirarchy` -> `hierarchy`

Reply:
Fixed, cheers :)